### PR TITLE
Fix enterprise user unsaved changes

### LINKED ISSUE #1216: UX confusing when adding/removing enterprise managers
I log in as an enterprise user, and I edit my enterprise and go to the "Users" tab. I enter the email address of a user I wish to add as a manager.

When I do so, the "Update" button is shown as active. But I haven't actually made a change to the form. I have to click the "+" button before the user will be added. **I expect the "Update" button to stay greyed out until I hit "+"**.

screenshot from 2018-03-09 15-06-36

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #1216.

Minor bugfix. Enterprise user form was showing "Unsaved Changes" message before a new user was properly added.

#### What should we test?

Adding new users to an enterprise. The unsaved changes bar shouldn't show before the user has been added to the list.

### CODE REVIEW

**Comment by sauloperez** [comment]
looks like CIs aren't happy

**Comment by myriamboure** [comment]
Happy to test @Matt-Yorkley as soon as your tests are passing. Waiting then ;-)
giphy

**Comment by Matt-Yorkley** [comment]
Tests are green now, @myriamboure :)

obligatory-gif

As a sidenote: Semaphore passed this time, and Travis needed a restart, which added 17 minutes to the CI build time...

**Comment by daniellemoorhead** [comment]
@myriamboure @Matt-Yorkley I applaud your use of animated gifs. Bravo! 😆

**Comment by pacodelaluna** [comment]
Deployed right now on french staging, https://staging.openfoodfrance.fr/

**Comment by RachL** [comment]
@myriamboure @Matt-Yorkley good for me :) Here are my testing notes : https://docs.google.com/document/d/1Z3dPAxhbRmGR5Qgq8U_IiU1PT3YRz_x7HGB7MeOwKi8/edit#

File: app/views/admin/enterprises/_ng_form.html.haml [hunk 1]
```diff
@@ -3,7 +3,6 @@
 -# So we use onchange and have to get the scope to access the ng controller
 = form_for [main_app, :admin, @enterprise], html: { name: "enterprise_form",
   "ng-controller" => 'enterpriseCtrl',
-  'onchange' => 'angular.element(enterprise_form).scope().setFormDirty()',
```
Comment (mkllnk) [inline app/views/admin/enterprises/_ng_form.html.haml]: I believe this one caused https://github.com/openfoodfoundation/openfoodnetwork/issues/2195.

Comment (myriamboure) [reply app/views/admin/enterprises/_ng_form.html.haml]: @Matt-Yorkley can you reopen a PR to fix that again then? I believe in v1.13 it' causing some issues in some instances so would be great to fix ASAP...

**Comment by myriamboure** [comment]
So something to learn here, even if not easy to know what part of the code can be impacted, it's good to do some sanity checks to the basics that could potentially be affected. Ping @RachL as well, I should have redone some tests as well probably, sounded pretty easy and straightforward but was not ;-)

File: app/views/admin/enterprises/_ng_form.html.haml (hunk 1 above)
Comment (myriamboure) [reply app/views/admin/enterprises/_ng_form.html.haml]: Oh seems @mkllnk fixed that in https://github.com/openfoodfoundation/openfoodnetwork/pull/2196 that was merged today so it seems it is done.

**Comment by enricostano** [comment]
> do some sanity checks to the basics

@myriamboure also, take note of those basic checks so we can make sure that we include them in the automated tests we run every PR get created.

File: app/views/admin/enterprises/_ng_form.html.haml (hunk 1 above)
Comment (myriamboure) [reply app/views/admin/enterprises/_ng_form.html.haml]: Oups nop sorry Maikel just did a PR to revert those changes so you still need to work on it @Matt-Yorkley :-(

**Pull request merged**